atomisticparsers/gromacs/parser.py

```diff
@@ -651,15 +652,17 @@ def parse_thermodynamic_data(self):
         calculation_times_ps = calculation_times_ps.magnitude * ureg.convert(
             1.0, calculation_times_ps.units, ureg.picosecond)
 
+        system_indices_assigned = []
         for n in range(n_evaluations):
```

Comment:
We can simplify this by looping over all sec_system. We convert calculation_times_ps to a mapping to get the thermo_data which we assign to the calculation. With this, we do not need line 696-722.

Reply:
If I understand correctly, your proposal does not take into account the situation when there are entries in the calculation section that do not correspond to a system. I still need to create a calculation entry there. Since the list in calculation and the list in system do not have to overlap at all, I don't see any way to avoid going through both lists here. If you still disagree, then I am missing something...?

Reply:
We can pop the values in thermo data, if there are still time steps left, then we create additional calculation sections in another loop. The question only is the sorting, but this can be done in a couple of lines, i guess.

Reply:
I think my idea is stupid sorry. So we loop over each thermo_data just the same and assign system_ref if time matched and pop the value from the system_time_map. If there are entries left in system_time_map, we loop over them and create addtional calculation sections.

Reply:
Is there a better way to keep track of the remaining entries in system_time_map, so that we don't need to loop over the full system list? 

I thought now of making an array of the system indices and then deleting the ones used in the thermo loop, but probably there is a better way.

Reply:
popping it like system_time.pop(time, None) will pop it if it exists other will return None.

Reply:
aha! I had no idea what you were referring to when you said "popping" before, which really added to my confusion. I did not know about this function. Ok, I will adjust the code, thanks!

Reply:
maybe we can still improve this if we concatenate all the times from calculation and system.